src/communication/NotificationSource.cpp

```diff
@@ -35,11 +39,19 @@ NotificationSource::NotificationSource(
 v1::UStatus NotificationSource::notify(
     datamodel::builder::Payload&& payload) const {
 	auto message = notify_builder_.build(std::move(payload));
+	if (!transport_) {
```

Comment:
Is this check necessary here? Is it possible to have a null transport_ member if it's checked before construction?

Reply:
It's not I agree. Removed